components/esp_wifi_remote/eppp/wifi_remote_rpc_server.cpp

```diff
@@ -32,7 +34,74 @@ const unsigned char key[] = "-----BEGIN PRIVATE KEY-----\n" CONFIG_ESP_WIFI_REMO
 
 using namespace server;
 
+struct Events {
+    api_id type;
+    int32_t id;
+    esp_wifi_remote_eppp_ip_event *ip_data{nullptr};
+    bool clean_ip_data{true};
+    esp_err_t create_ip_data()
+    {
+        ip_data = new (std::nothrow) esp_wifi_remote_eppp_ip_event;
+        return ip_data ? ESP_OK : ESP_ERR_NO_MEM;
+    }
+    ~Events()
```

Comment:
Could you explain the usage you expect here? The `struct` has only default constructed data and depending on state the destructor might leak the allocated `ip_data`.

Reply:
The idea is using RAII even when en-queuing and later de-queuing the instance.

> the destructor might leak the allocated ip_data.

* we construct `ip_data` only when needed (if we get the appropriate `IP_EVENT`)
* we put this instance to the queue and only after (successful) en-queuing we mark this as `DO-NOT-FREE`, so the desctructor *leaks* the data, as they will be used afterwards...
* we get the instance from the queue with the default state, so now the destructor *free*s the data as expected

(briefly commented here https://github.com/espressif/esp-protocols/pull/582/files#diff-e2d9eebe2ae3236f3c8b216ed4f69dd878089e5f7c1b2fe3fb9a23c238651b8fR61)